mava/distributions.py

```diff
@@ -91,6 +92,61 @@ def _parameter_properties(cls, dtype: Optional[Any], num_classes: Any = None) ->
         return td_properties
 
 
+class MaskedEpsGreedyDistribution(tfd.Categorical):
+    """
+    Computes an epsilon-greedy distribution for each action choice. There are two
+    components in the distribution:
+
+    1. A uniform component, where every action that is NOT masked out gets an even weighting.
+    2. A greedy component, where the action with the highest corresponding q-value that is
+    NOT masked out gets a probability of one.
+
+    Combining these two distributions per action choice in a ratio of eps:1-eps gives
+    the final distribution. This distribution can be sampled using mode() for a purely
+    greedy strategy, and sampled normally using sample() for an epsilon-greedy strategy.
+    """
+
+    def __init__(self, q_values: chex.Array, epsilon: float, mask: chex.Array):
+        self.q_values = q_values
```

Comment:
nice to explain why we're keeping them
```suggestion
        # keep q values available if we need to use them to learn with later
        self.q_values = q_values
```